### PR TITLE
Initial tests suite & factories

### PR DESCRIPTION
**Pull request by GeniJaho**
- Fixed PHPUnit config file (can't use SQLite)
- Tests use a MySQL database called **olm_test** on the same DB connection
- By default, all tests use the RefreshDatabase trait, included in the base test class
- Added .idea to .gitignore, those files keep showing in diffs
- Unit tested the Photo model

I noticed that Coastal and Art models need a photo_id to be created, but the others don't. Is that ok, or they shouldn't need it either?
Also, the Photo model has Country, State, City both as relationships and DB fields. The relationships are wrong and they don't work, instead of **hasOne** they should be **belongsTo**. Are these relationships used elsewhere in the project? Can we delete them?

### CODE REVIEW

**Comment by xlcrr** [comment]
Thanks!

photo_id can be removed from Coastal and Art. Storing the 1-way relationship from photo->$category is enough

I think we can delete them

File: app/Models/Photo.php [hunk 1]
```diff
@@ -173,13 +176,8 @@ public function total ()
...
             {
                 // We dont want to include brands in total_litter
-                if ($category === 'brands')
-                {
-                    $this->$category->total();
-                }
-
                 // Increment total_litter when its not brands
-                else
+                if ($category !== 'brands')
```
Comment (xlcrr) [inline app/Models/Photo.php:180]: We still need to call the first block, which calculates and saves `total_brands` on the brands model.

The second block increments `total_litter`, which is a sum of all litter, expect brands

We could brands and litter separately which is why this block is needed here

File: database/factories/Litter/Categories/AlcoholFactory.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'beerCan' => $this->faker->randomDigit
```
Comment (xlcrr) [inline database/factories/Litter/Categories/AlcoholFactory.php:25]: You can do `$this->$category->types()` on any money to get a list of keys

File: tests/Unit/Models/PhotoTest.php [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            'brands_id' => $brands->id
+        ]);
+
+        // Suppress output to console
+        $this->setOutputCallback(function() {});
+
+        $photo->total();
+
+        // Brands are not calculated
+        $this->assertEquals($smoking->total(), $photo->total_litter);
```
Comment (xlcrr) [inline tests/Unit/Models/PhotoTest.php:103]: We should count total_brands separately

**Review by xlcrr — CHANGES_REQUESTED** [review]

File: app/Models/Photo.php (hunk 1 above)
Comment (GeniJaho) [reply app/Models/Photo.php:180]: Actually $this->$category->total() does not do anything on the LitterCategory models, it doesn't increase their counter. Or am I missing something?
Anyway I'm setting the method's signature as it was before.

File: database/factories/Litter/Categories/AlcoholFactory.php (hunk 2 above)
Comment (GeniJaho) [reply database/factories/Litter/Categories/AlcoholFactory.php:25]: I actually wanted to keep the factories separate from app logic. How can I use $this->$category->types() in a factory? The types() method is not static, and we don't have a model inside the factory?

File: database/factories/PhotoFactory.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        return [
+            'user_id' => User::factory()->create(),
```
[the remark this answers is not in the document]
Comment (GeniJaho) [reply database/factories/PhotoFactory.php:26]: We don't need to provide an id. Actually just checked the docs and we can go a step further 😂 
```
    return [
        'user_id' => User::factory(),
        'title' => $this->faker->title(),
        'content' => $this->faker->paragraph(),
    ];
```

File: tests/Unit/Models/PhotoTest.php [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        $photo = Photo::factory()->create();
+
+        $annotation = Annotation::factory()->create([
+            'photo_id' => $photo->id
+        ]);
+
+        $this->assertInstanceOf(Collection::class, $photo->boxes);
+        $this->assertCount(1, $photo->boxes);
+        $this->assertTrue($annotation->is($photo->boxes->first()));
+    }
```
[the remark this answers is not in the document]
Comment (GeniJaho) [reply tests/Unit/Models/PhotoTest.php:61]: I don't quite understand. The relationship is a hasMany, that's why I called it _has_many_boxes_. True, I only pass it only one box, but I assert that it's a Collection, not an Annotation instance.

How do we test for only one box? You mean we add a new box() relationship on the model?

File: app/Models/Photo.php (hunk 1 above)
Comment (xlcrr) [reply app/Models/Photo.php:180]: Actually you are right, this line can be deleted. `total()` returns a total count of all items on that modal, which is used in some places but not here. Your original code was correct and can be accepted

File: database/factories/Litter/Categories/AlcoholFactory.php (hunk 2 above)
Comment (xlcrr) [reply database/factories/Litter/Categories/AlcoholFactory.php:25]: Leave it for now, we will look into making them static in another PR

File: tests/Unit/Models/PhotoTest.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (xlcrr) [reply tests/Unit/Models/PhotoTest.php:61]: I was thinking we could do one test for `assertCount(1)` and another one for `assertCount(n)` but this is okay for now

**Pull request merged**